### DISCUSSION TITLE
How to build gradle project in termux?

### DISCUSSION THREAD

**Discussion by manjarolinuxuser** [termux/termux-packages#17725] [Q&A] [answered]
I want to build an app available on github

https://github.com/MoKee/android_packages_apps_WarpShare/tree/mkq-mr1

I am a newbie and I don't know to how to build this app

I want to build it using termux

Can someone tell me how to do it from scratch

Thanks in advance

**Comment by twaik** [comment]
Every gradle project must have `build.gradle`, `settings.gradle`, `gradlew` (optional) and `graldew.bat` files in its root. That project you mentioned does not have them...

**Reply by sylirre** [reply, marked as answer]
The mentioned project expected to be compiled as part of Android source code tree. Such approach doesn't use gradle as all dependencies are present within the OS source code and build flow is fully controlled by Android.mk file.

Answering the original question: it is not possible to build this particular app in Termux without significant changes.

For gradle projects: we already have the `gradle` package.

**Reply by manjarolinuxuser** [reply]
So can I know how to build this project. I can use a PC (Both windows and linux).

**Reply by twaik** [reply]
You can write build.gradle script for this project.

**Reply by manjarolinuxuser** [reply]
build.gradle is there 
See in MaterialProgressBar

**Reply by twaik** [reply]
It is related only to MaterialProgressBar, not to other parts of project. There is no way to build it in it's current form using only gradle.

**Reply by manjarolinuxuser** [reply]
I am new and I haven't compiled any apps yet
